core/rio/turtle/src/main/java/org/eclipse/rdf4j/rio/turtle/TurtleParser.java

```diff
@@ -1254,13 +1254,16 @@ protected void unread(int codePoint) throws IOException {
 	 * @throws IOException
 	 */
 	protected void unread(String string) throws IOException {
-		for (int i = string.codePointCount(0, string.length()); i >= 1; i--) {
+		int i = string.codePointCount(0, string.length());
+		while ( i > 0 ) {
 			final int codePoint = string.codePointBefore(i);
 			if (Character.isSupplementaryCodePoint(codePoint)) {
 				final char[] surrogatePair = Character.toChars(codePoint);
 				reader.unread(surrogatePair);
+				i -= surrogatePair.length();
```

Comment:
Good catch, thanks!

Reply:
@jeenbroekstra I added a test, which sadly shows this still fail, and fix was not enough.
I'm not sure where the problem lies. Also not sure If I'll have time to continue debugging this.
Will you be able to take this from here?

Reply:
No problem - though can you still make sure you sign the ECA so we can continue from your work? Otherwise, I'm afraid we'll have to drop this and start from scratch.

Reply:
@hochgi Oh hang, I see you've signed off and that your ECA has been completed as well, so that's all good. 

It's currently failing because this PR also contains a commit by @elad-shaked, who has signed off, but his e-mail address (elad.shaked@sparkbeyond.com) is not recognized by the ECA verifier. @elad-shaked if you could sign the ECA that'd be great. Otherwise, since your commit is only formatting-related related, I can work around it by cherrypicking the other fixes onto a new branch.

Reply:
as per @elad-shaked's request I just squashed his changes into mine.

Reply:
@jeenbroekstra just committed a fix suggested by @elad-shaked, and this seems to work :+1:

Reply:
@jeenbroekstra running `mvn test` from `core` passes locally but fails in CI :confused: